### ISSUE TITLE
Expressions containing stochastic processes are not simplifiable

### ISSUE THREAD

**Issue by Maelstrom6** [sympy/sympy#20078]
I have a concern for two of the operations related to `sympy.stats.stochastic_process_types.py`.

The two operations are simplification and differentiation.

```python
from sympy import *
from sympy.stats import *

# other SPs should be similar
W = WienerProcess('W')
t = symbols('t', positive=True)

# AttributeError: 'function' object has no attribute 'is_Rational'
print((2*W(t)).simplify())

# should raise an error
print(W(t).diff(t))
```
Simplification seems to fail when called on an expression containing a stochastic process. `W(t).simplify()` works though.

The Wiener Process is nowhere differentiable with probability 1. I think differentiation should raise an error in that case.
There are many sources on this but here are some quick ones. [1](https://en.wikipedia.org/wiki/Wiener_process) [2](https://quant.stackexchange.com/questions/10861/how-can-the-wiener-process-be-nowhere-differentiable-but-still-continuous) [3](http://www.math.uchicago.edu/~may/VIGRE/VIGRE2009/REUPapers/McKnight.pdf).

A bit off-topic, but I also feel that the core of stats such as `rv.py` where designed for uni-variate distributions. At least that's why I think there are many issues related to joint distributions. Maybe `rv.py` and `joint_rv.py` and others should be redesigned with this is mind. It would be great to have an ML estimator approach and hopefully be able to incorporate known theorems with expressions of random variables. For example, that the sample mean and sample variance are independent for any normal random variable.

**Comment by sylee957** [comment]
There can be little to none implementations of `simplify` applied for expressions involving random variables.
I'm not sure if you can simplify `2*W(t)` to give any meaningful result,
However, as a lot of random variable expressions follow some different algebraic rules, there should be a survey about what kind of canonical forms can be taken for expressions like `E(X**2) - E(X)**2`

**Comment by oscarbenjamin** [comment]
The issue with `simplify` is that `WeinerProcess` has messed up args:
```julia
In [7]: W(t).args
Out[7]: (W(t), StochasticPSpace(W, WienerProcess(W), <bound method WienerProcess.distribution of WienerProcess(W)>))
```
All args should be instances of `Basic` so there shouldn't be a bound method in there.

**Comment by Maelstrom6** [comment]
> I'm not sure if you can simplify `2*W(t)` to give any meaningful result

No, I don't think you can either. I was hoping to get back `2*W(t)`.


> The issue with `simplify` is that `WeinerProcess` has messed up args

Ah, all right. I'll do a bit of digging around to check if the method could rather exist inside a Basic object rather than its args.

**Comment by czgdp1807** [comment]
> The issue with `simplify` is that `WeinerProcess` has messed up args:

W(t) is a `RandomIndexedSymbol`. The first argument is an `Indexed` object and the second is `StochasticPSpace`, and in `StoachasticPSpace` the first argument is symbol, the second is the process and the third is the distribution. Now, instead of the actual distribution, the method is placed because `self.distribution()` was never called (see below). This should be fixed.

https://github.com/sympy/sympy/blob/089397d2e4027dab3560d3c187f98bfee6f41fc3/sympy/stats/stochastic_process_types.py#L254

**Comment by czgdp1807** [comment]
Proposed fixes, https://gist.github.com/czgdp1807/879223647496d9cef0b246272731c3db
It involves some workarounds to minimise the changes.

**Comment by 0sidharth** [comment]
@czgdp1807 Should I open a PR with the proposed fixes? what work is left in this apart from testing?

**Comment by czgdp1807** [comment]
Sure. I really don't remember what all is left to do. Though you can still open a PR by applying the fix and adding more tests.
